app/content/react/library-reference/widgets.md

```diff
@@ -59,6 +62,57 @@ Renders a `<CategoryWidget />` component, binded to a source at redux. The widge
   // The operationColumn wouldn't be required if using AggregationTypes.COUNT, to count the number of countries per continent
   ```
 
+#### FeatureSelectionWidget
+
+Renders a `<FeatureSelectionWidget />` component. The widget allows the user to draw a shape on the map and apply a filter to select the features within the shape. Once a shape has been drawn, it can be selected and modified by adding/removing vertices or translated to a new location. By default the mask is active but it can be disabled temporarily and re-enabled again.
```

Comment:
This paragraph seems to say that the widget is responsible of drawing the shape, when it's just for selecting the mode.

> The widget allows the user to draw a shape on the map and apply a filter to select the features within the shape.

Reply:
Hahaha. Well, maybe I would say that FeatureSelectionWidget allows to select the mode that will be used by the FeatureSelectionLayer to draw the shape. I'd indicate that we differ between drawing modes (each one draws a different shape) and edition mode (that allows to modify a drawed shape).
But it's my perception of this paragraph, maybe I'm worng.

Reply:
You are right about the responsibility but my intention here is to describe the widget functionality, not the implementation details.

Reply:
But it works like that using with the layer. By itself it only select the mode. From my knowledge, people don't read too much haha. If I just read this line, I would think that if I add FeatureSelectionWidget, it just works haha. But I understand your point, it's good.